texar/torch/run/executor.py

```diff
@@ -1483,6 +1493,37 @@ def erase_status_and_log_fn(executor: 'Executor'):
             self.test_log_format, self.test_progress_log_format,
             Event.TestingIteration, Event.Testing)
 
+    def _register_tbx_logging_actions(self):
+
+        # Register logging actions.
+        Points = Sequence[Union[Condition, Event]]
+        LogFn = Callable[['Executor'], str]
+
+        def _register(points: Points, fn: ActionFn):
+            for cond_or_event in points:
+                if isinstance(cond_or_event, Condition):
+                    self._register_action(cond_or_event, fn)
+                else:
+                    self._register_hook((cond_or_event, True), fn)
+
+        def tbx_train_log_fn(executor: 'Executor'):
+            train_metrics = executor.train_metrics
+
+            # log the metrics here
+            for key, value in train_metrics.items():
+                self.summary_writer.add_scalar(
+                    f"train/{key}", value.value(), executor.status["iteration"])
+
+        def tbx_eval_log_fn(executor: 'Executor'):
+            valid_metrics = executor.valid_metrics
+
+            for key, value in valid_metrics.items():
+                self.summary_writer.add_scalar(
+                    f"eval/{key}", value.value(), executor.status["epoch"])
+
+        _register(self._tbx_logging_conditions, tbx_train_log_fn)
+        _register([Event.Epoch], tbx_eval_log_fn)
```

Comment:
Validation is performed on `validation_every` conditions, it's not necessarily per epoch. It's also possible to perform validation multiple times in an epoch (e.g., the Transformer example). I think we can also use iteration as the X-axis value.

Reply:
And, instead of `eval`, maybe we can just call it `valid`. To me `eval` feels like the union of valid and test.

Reply:
Not really. The iteration counts from the beginning of training and does not reset after each epoch. So if `executor.status["iteration"] == 10000` after the first epoch, it will become `30000` after the third epoch.

You can also add entries to the `status` dictionary as you wish, just make sure you document it. But I feel it's better to use iteration rather than validation counts, because it's possible that validation is performed at non-uniform intervals.